07-On demand scan as an API/README.md

```diff
@@ -0,0 +1,252 @@
+
+9. When the secret is created, note the key value for use in a subsequent step.
+
+</br>
+
+**Note for WebAPI Admin**
+> Enable the flag 'OnDemandScanAPI' in order to provide access over API endpoints.
+
+[Back to top…](README.md#On-this-page)
+
+<!-- TBD: This token is not access subs, but to access AzTS REST API endpoints -->
+
+## Generate user authentication token to access subscriptions
+User has to generate authentication token in order to use APIs for any subscription.
+There are two ways to generate access tokens:
+- Using client credential flow
+- Using user authentication code flow
+
+#### Required Az Module
+``` PowerShell
+Install-Module -Name MSAL.PS -AllowClobber -Scope CurrentUser -repository PSGallery
+```
+
+### Required roles:
+You must have permission over a subscription with any of the following role:
+- Owner
+- Contributor
+- ServiceAdministrator
+- CoAdministrator
+- AccountAdministrator
+- Security Reader
+- Security Admin
+> **Note:** You need to run RBAC after granting the permission to SPN over subscription.
+
+### Using client credential flow
+Client crediential flow uses the client credentials (client id and client secret) to generate the token. Token will be generated against specified SPN (Service Principal Name) and **SPN must have access over the subscription** to scan or to get the control scan result.
+
+> In order to generate the token for APIs, you have to get access for the client application from WebAPI owner.
+> 1. Send the client id to WebAPI owner to request access for client application.
+> 2. WebAPI owner will grant the access and share the scope.
+> 3. Use WebAPI scope while generating the access token.
+
+**Steps for WebAPI owner to get scope:**
+> 1. Go to Azure Portal.
+> 2. Go to App Registration.
+> 3. Select WebAPI App Registration.
+> 4. Go to 'Expose an API'.
+> 5. Copy scope from 'Scopes'.
+> </br>
+
+Commands to generate the token:
+``` PowerShell
+
+$ClientSecret = '<client-secret>' | ConvertTo-SecureString -AsPlainText -Force
+
+$token = Get-MsalToken -TenantId '<tenant-id>' -ClientId '<client-id>' -ClientSecret $ClientSecret -Scopes "<WebAPI-scope>/.default"
+
+$header = "Bearer " + $token.AccessToken
+
+$headers = @{"Authorization"=$header;"Content-Type"="application/json";}
+
+```
+
+[Back to top…](README.md#On-this-page)
+
+### Using user authentication code flow
+User authentication code flow uses user's crediential to generate the token. User must have access over the subscription  to scan or to get the control scan result.
+
+<!-- TBD: Also, 'Admin consent' is required only when the access of granting consent is restricted to admin at AD level -->
+
+**Note for WebAPI Owner:**
+> WebAPI must have 'Admin consent' granted to expose the APIs.
+> </br>
+> Grant admin consent for client app registration:
+> 1. Go to Azure Portal.
+> 2. Go to 'App Registration'.
+> 3. Select appropriate 'App Registration'.
+> 4. Go to 'API Permissions'.
+> 5. Click 'Grant admin consent for Microsoft'.
```

Comment:
You should mention the scope for which consent has to be granted.

Reply:
Done